feat : Comment Create Read 기능 개발

jade-coding opened the pull request:
## PR Summary

- 댓글을 Create하여 DB에 저장하는 기능을 개발했습니다. 
- resolves #1 

## Changes

- Comment 엔티티 생성
- Comment 리포지토리 생성
- Spring Data Jpa가 제공하는 기능 사용
- Comment 서비스 생성
- 댓글 저장 기능 구현 및 테스트
- Post <-> Comment 일대다 맵핑 
- 엔티티 필드 수정
- 테스트 파일 미 사용 코드(beforeEach함수) 제거
- 댓글 ID로 댓글 조회 기능
- Post ID로 댓글 조회 기능


## Test Checklist
- [x] 댓글 저장 테스트 성공
- [x] 댓글 조회 테스트 성공

jade-coding committed:
feat : 댓글 엔티티 생성

jade-coding committed:
feat : 댓글 리포지토리 생성

jade-coding committed:
feat : 댓글 저장 기능 구현

jade-coding committed:
test : 댓글 저장 기능 테스트

jade-coding committed:
feat : Post, Comment 일대다 연관관계 맵핑  

- Comment 엔티티에 Post 엔티티와 맵핑
- 생성자 호출시 Post가 주입

jade-coding committed:
chore : Comment 엔티티 필드명 수정

- createAt -> createdAt 변경
- updateAt -> updatedAt 변경

jade-coding committed:
test : Comment Create 기능 테스트 수정

- 변경 전) comment post 맵핑 없이 DB에 저장  
- 변경 후) comment - post 맵핑 후 comment에 post가 주입되어져서 DB에 저장

jade-coding committed:
chore : Comment 엔티티 필드 수정 

- title, imgUrl, author 필드 삭제
- contents -> content 필드명 변경
- commentWriter 추가

jade-coding committed:
chore : Comment 엔티티 필드 수정

- content -> contents 수정
- commentWriter -> author
- updatedAt -> updateAt
- createdAt -> createAt

jade-coding committed:
test : CommentServiceTest  @beforeEach 코드 제거

- @beforeEach, beforeEach함수 삭제 - 미 사용

jolly-n commented on back/blog/src/main/java/saramdle/blog/domain/Comment.java:
```diff
@@ -18,17 +18,17 @@ public class Comment {
-    private LocalDateTime updatedAt;
+    private LocalDateTime updateAt;
 
     @Builder
```
생성자의 변수명은 변경이 안된거 같습니다!

an2z reviewed (changes requested):
필드명 변경해주신게 생성자에는 적용이 안된거 같아 확인 후 수정부탁드립니다.

an2z commented on back/blog/src/main/java/saramdle/blog/domain/Comment.java:
```diff
@@ -0,0 +1,43 @@
+        this.commentWriter = commentWriter;
+        this.createdAt = createdAt;
+        this.updatedAt = updatedAt;
+    }
```
생성자에 변수명도 같이 변경되어야 할 것 같습니다.

jade-coding committed:
chore : Comment 엔티티 필드 수정

- content -> contents 변경
- updatedAt, createdAt -> updateAt, createAt 변경
- commentWriter -> author 변경

jade-coding committed:
feat : 댓글ID로 댓글 조회 기능 구현

- 댓글 고유 ID에 따른 댓글 데이터 조회 기능
- Null일 경우 NOT_FOUND_COMMENT 에러 발생처리

jade-coding committed:
test : 댓글ID로 댓글 조회 기능 테스트

jade-coding committed:
feat : Post ID로 댓글 조회 기능 구현

- JPA가 인터페이스에 선언된 findByPost_id에 대하여 구현 객체를 동적으로 생성하여 쿼리 조건에 맞는 데이터들 반환
-  조건에 맞는 데이터들 List 형태로 반환
- List가 비어있는 경우  NOT_FOUND_COMMENT 에러 발생

jade-coding committed:
test : Post ID로 댓글 조회 기능 테스트

jade-coding committed:
test : beforeEach() 구문 삭제

jade-coding committed:
chore : Comment 엔티티 필드 수정

-createAt, updateAt -> createdAt, updatedAt

an2z commented on back/blog/src/test/java/saramdle/blog/service/CommentServiceTest.java:
```diff
@@ -43,15 +42,29 @@ void save() {
+
+        commentService.save(comment);
+
+        List<Comment> findedComments = commentService.findComments(comment.getPost().getId());
```
위에서 `Long postId = postService.save(post)`로 postId를 변수로 추출하여
`commentService.findComments(postId);` 이런식으로 해당 변수를 재사용하면 좋지 않을까요?

an2z commented on back/blog/src/test/java/saramdle/blog/service/CommentServiceTest.java:
```diff
@@ -43,15 +42,29 @@ void save() {
+
+        List<Comment> findedComments = commentService.findComments(comment.getPost().getId());
+
+        assertThat(findedComments.size()).isEqualTo(1);
```
`assertThat(findedComments).hasSize(1)` 이렇게 컬렉션의 사이즈를 비교하는 기능을 따로 제공해주는데,
해당 기능을 활용한다면 가독성이 더욱 좋아질거 같아요.

an2z commented on back/blog/src/test/java/saramdle/blog/service/CommentServiceTest.java:
```diff
@@ -0,0 +1,64 @@
+    @Autowired
+    PostService postService;
+
+
```
라인 하나 지워주셔도 될것 같아요.

jade-coding committed:
chore : 변수명 맞춤법 수정

- findedComments -> foundComments

jade-coding committed:
refactor : Post ID로 댓글 조회 코드 에러 처리 삭제

- 댓글 조회시 해당 Post ID에 대한 댓글이 존재하지 않을 경우 빈 리스트를 반환해주므로 비었을 경우에 대한 에러처리가 불 필요해서 코드 제거
